flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -684,6 +689,96 @@ private void setCoLocation() {
 		}
 	}
 
+	private void setManagedMemoryFraction() {
+		// all slot sharing groups in this job
+		final Set<SlotSharingGroup> slotSharingGroups = Collections.newSetFromMap(new IdentityHashMap<>());
+		// maps a job vertex ID to its head operator ID
+		final Map<JobVertexID, Integer> vertexHeadOperators = new HashMap<>();
+		// maps a job vertex ID to IDs of all operators in the vertex
+		final Map<JobVertexID, Set<Integer>> vertexOperators = new HashMap<>();
+
+		for (Entry<Integer, JobVertex> entry : jobVertices.entrySet()) {
+			final int headOperatorId = entry.getKey();
+			final JobVertex jobVertex = entry.getValue();
+
+			final SlotSharingGroup jobVertexSlotSharingGroup = jobVertex.getSlotSharingGroup();
+			if (jobVertexSlotSharingGroup != null) {
```

Comment:
Under what circumstances would this not be the case at this point? If this case can occur, then a test is missing.

Reply:
Currently there is no way to set it null in production. `StreamJobGenerator` would always assign a non-null slot sharing group to `StreamNode`, and that would be the group of the generated `JobVertex`. (For the case of DataSet all vertices are in the "default" group.)

However, the nullable assumption of slotSharingGroup is respected in many places, including @Nullable markers in many places and single slot allocation code paths in `SchedulerImpl`.

Dealing with job vertex with null slot sharing group does not make much sense currently, and we can hardly test it in normal ways.
How about keeping it simple by just adding a check to ensure the job vertex slot sharing group is not null? And also open a follow up issue to determine whether to support null slot sharing group in runtime?

Reply:
sounds good.

Reply:
FLINK-14870 is opened to track this issue.